### PR TITLE
Add ros2_control_solo_bringup, ros2_control_test_nodes, ros2_description_solo packages

### PR DESCRIPTION
**Pull request by swuuu**
Initial PR for SOLO 12.
@olivier-stasse, @Lhumd, @MaximilienNaveau

### CODE REVIEW

**Review by olivier-stasse — CHANGES_REQUESTED** [review]
Dear @swuuu,
Thanks a lot for your massive contribution.
I am very happy to see that the bolt repository was helpful.

Your addition in the test_node is super impressive, you actually implemented a whole controller.
For this very reason, I do not think that putting it in a ros_test_node is very appropriate.

The current aim with this repository is to be independent of the controller.

We have at least two to three students working on controlling SOLO with various control architectures.  

It would be nice to put it on another package. If this is related to a publication or a master student work you would highlight your contribution to the field by having it clearly separated.


So to summarize please put ros2_control_test_nodes in a different repository with the name corresponding to the control architecture you implemented.

There is also an issue with the interfaces available on the real robot and Gazebo (Kp/Kd Gains), we wrote a Gazebo plugin for that, but this is  still embedded in the bolt repository.

Again thanks a lot for this very nice contribution.

**Comment by Lhumd** [comment]
Hi,

Thanks a lot for your helpful comment, @olivier-stasse. I agree. It's better to have a separate package for controllers in ros2 like mim_control. Also, merging some parts of the current repository to reactive_planner could be a good idea as those are not simple controllers.
 
Our group is going to use ros2 and ros2_control for Solo12 to skip the complexity of dynamic_graph. We appreciate your suggestion and comments. 

Please, let us know if there are similar projects on your side that can help us. And I hope @swuuu's work would be useful for your students too.

BTW, we didn't finish building Solo12 here. So these codes never checked on the real robot.

**Comment by olivier-stasse** [comment]
> Hi,
> 
> Thanks a lot for your helpful comment, @olivier-stasse. I agree. It's better to have a separate package for controllers in ros2 like mim_control. Also, merging some parts of the current repository to reactive_planner could be a good idea as those are not simple controllers.
> 
> Our group is going to use ros2 and ros2_control for Solo12 to skip the complexity of dynamic_graph. We appreciate your suggestion and comments.

I understand. The chainable part of 'dynamic_graph' is currently implemented in the ros2_control framework. And the community will benefit from that.

> Please, let us know if there are similar projects on your side that can help us. And I hope @swuuu's work would be useful for your students too.

His contribution is definitely useful. I will use it to have this repository using the odri hardware.

> BTW, we didn't finish building Solo12 here. So these codes never checked on the real robot.

Thanks for specifying it. I was clear from @swuuu comments.
I propose that, once the controller is in a different package we merge this PR. Then we upgrade it with the odri interface and possibly the gazebo plugin which goes with it.
Will that be fine for your team ?

**Comment by swuuu** [comment]
Dear @olivier-stasse and @Lhumd ,

The ros2_control_test_nodes package has been removed. The launch files launching the nodes in ros2_control_test_nodes has also been removed.

Thank you!

**Comment by swuuu** [comment]
Dear @olivier-stasse,

> Thanks for specifying it. I was clear from @swuuu comments. I propose that, once the controller is in a different package we merge this PR. Then we upgrade it with the odri interface and possibly the gazebo plugin which goes with it. Will that be fine for your team ?

Apologies for the late response. Yes, our group agrees with your suggestion.

Thank you for your time.

**Comment by olivier-stasse** [comment]
Sorry it took me so long to merge.
Thanks again for the great contribution.
